keras_cv/layers/object_detection/retina_net_label_encoder.py

```diff
@@ -91,39 +105,23 @@ def _match_anchor_boxes(
         iou_matrix = bounding_box.compute_iou(
             anchor_boxes, gt_boxes, bounding_box_format="xywh"
         )
-        max_iou = tf.reduce_max(iou_matrix, axis=1)
-        matched_gt_idx = tf.argmax(iou_matrix, axis=1)
-        positive_mask = tf.greater_equal(max_iou, match_iou)
-        negative_mask = tf.less(max_iou, ignore_iou)
-        ignore_mask = tf.logical_not(tf.logical_or(positive_mask, negative_mask))
-        return (
-            matched_gt_idx,
-            tf.cast(positive_mask, dtype=self.dtype),
-            tf.cast(ignore_mask, dtype=self.dtype),
-        )
-
-    def _encode_sample(self, gt_boxes, anchor_boxes):
```

Comment:
Not strongly opinionated, but curious: do we need to merge this and  `_match_anchor_boxes` into one method?

IMO It's a little easier to read when broken up

Reply:
I thought this removes `_match_anchor_boxes` entirely?

Reply:
ack -- yeah I don't mind removing the API surface, just thought this new `_encode_sample` method is now doing a lot.
But if you and Luke are happy with it then sgtm

Reply:
It's a little messy mostly due to extra updating a metrics. Otherwise the core logic here is:
1. compute_iou
2. re-using box_matcher to get matched results
3. map matched results to box targets and class targets